src/Diagnostics.RuntimeHost/Extensions/EncryptedConfigProviderExtensions.cs

```diff
@@ -0,0 +1,101 @@
+﻿using Diagnostics.Logger;
+using Microsoft.Extensions.Configuration;
+using Newtonsoft.Json;
+using System;
+using System.Collections.Generic;
+using System.IO;
+using System.Security.Cryptography;
+using System.Text;
+
+namespace Microsoft.Extensions.DependencyInjection
+{
+    public static class EncryptedConfigProviderExtensions
+    {
+        public static IConfigurationBuilder AddEncryptedProvider(this IConfigurationBuilder builder, string encryptionKey, string initializationVector, string file)
+        {
+            return builder.Add(new EncryptedConfigProvider(encryptionKey, initializationVector, file));
+        }
+    }
+
+    public class EncryptedConfigProvider : ConfigurationProvider, IConfigurationSource
+    {
+        static string EncryptionKey;
+
+        static string EncryptedFile;
+
+        static string InitializationVector;
+
+        public EncryptedConfigProvider()
+        {
+
+        }
+
+        public EncryptedConfigProvider(string configEncryptionKey, string configInitVector, string file)
+        {
+            EncryptionKey = configEncryptionKey;
+            EncryptedFile = file;
+            InitializationVector = configInitVector;
+        }
+
+        public override void Load()
+        {
+            Data = DecryptSettings();           
+        }
+
+        private IDictionary<string, string> DecryptSettings()
+        {
+            var config = new Dictionary<string, string>();
+            try
+            {
+                using (StreamReader reader = new StreamReader(EncryptedFile))
+                {
+                    string json = reader.ReadToEnd();
+                    var result = JsonConvert.DeserializeObject<Dictionary<string, string>>(json);
+
+                    foreach (KeyValuePair<string, string> item in result)
+                    {
+                        var plainText = DecryptSetting(item.Value);
+                        if (!string.IsNullOrEmpty(plainText))
+                        {
+                            config.Add(item.Key, plainText);
+                        }
+                    }
+
+                }
+            } catch (Exception ex)
+            {
+                DiagnosticsETWProvider.Instance.LogRuntimeHostMessage($"Exception in {nameof(EncryptedConfigProvider)} : {ex.ToString()}");
```

Comment:
if Decryption of the setting fails, do we want to return gracefully? It might cause failures in downward streams.
Should we look to :
a) throw the exception here with exception details getting logged into kusto
b) let the process not start in this condition, so that health check/ front door etc can kick in and save our PROD SLA
c) put an alert on this above exception with TSG?

Reply:
Yes, those steps make sense. We can implement those.